c2_components/include/mfx_c2_decoder_component.h

```diff
@@ -212,7 +220,8 @@ class MfxC2DecoderComponent : public MfxC2Component
     std::list<std::unique_ptr<C2Work>> m_flushedWorks;
 
     std::mutex m_readViewMutex;
-    std::map<decltype(C2WorkOrdinalStruct::timestamp), std::unique_ptr<C2ReadView>> m_readViews;
+    std::map<const uint64_t, std::unique_ptr<C2ReadView>> m_readViews;
+    std::list<std::pair<uint64_t, uint64_t>> m_duplicatedTimeStamp;
```

Comment:
I think you don't need to create a new list to prevent a duplicated timestamp frame. You just need to modify the type of m_readViews like `std::map<std::pair<int, int>, std::unique_ptr<C2ReadView>> m_newReadViews;` and modify find lambda like 
`            auto it = find_if(m_newReadViews.begin(), m_newReadViews.end(), [timestamp, index](const auto &item){
                return item.first.first == timestamp && item.first.second == index;
            });`
This will make the code simpler and clearer.

Reply:
Need this list seperately. It is convenient and flexible for us to manage the duplicated timestamp.